app/models/navigation_link.rb

```diff
@@ -0,0 +1,8 @@
+class NavigationLink < ApplicationRecord
+  URI_REGEXP = URI::DEFAULT_PARSER.make_regexp(%w[http https]).freeze
+  SVG_REGEXP = /\A<svg .*>/i.freeze
+
+  validates :name, :url, :icon, presence: true
+  validates :url, format: URI_REGEXP
```

Comment:
why not use the URI validator 😊 ?

For example:

```ruby
validates :url, url: { schemes: %w[https http] }
```

Reply:
✅